logo.js

```diff
@@ -76,6 +76,10 @@ function LogoInterpreter(turtle, stream, savehook)
     return atom === match;
   }
 
+  function isPromise(value) {
```

Comment:
In ECMAScript parlance, the duck-type is called a "thenable" and the existence of `then` is sufficient to identify it. Even in the ECMAScript spec itself when e.g. passing to `Promise.resolve()`

So maybe `isThenable`